src/builder.rs

```diff
@@ -942,16 +1048,17 @@ impl<V> Bundle<InProgress<Proven, PartiallyAuthorized>, V> {
     ///
     /// Returns an error if any signatures are missing.
     pub fn finalize(self) -> Result<Bundle<Authorized, V>, Error> {
-        self.try_map_authorization((
-            Ok,
-            Ok,
-            |maybe: MaybeSigned| maybe.finalize(),
-            |partial: InProgress<Proven, PartiallyAuthorized>| {
+        self.try_map_authorization::<(), _, _>(
+            (),
```

Comment:
Is the type annotation necessary here? Can `rustc` not figure out the type of `R` from the first argument?
```suggestion
        self.try_map_authorization(
            (),
```